[2023-08-29T22:03:32.670+0000] {processor.py:157} INFO - Started process (PID=24040) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:03:32.672+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:03:32.674+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:03:32.673+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:03:32.725+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:03:32.717+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:03:32.726+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:03:32.762+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.095 seconds
[2023-08-29T22:04:03.036+0000] {processor.py:157} INFO - Started process (PID=24135) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:04:03.038+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:04:03.039+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:04:03.039+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:04:03.082+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:04:03.074+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:04:03.083+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:04:03.127+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.096 seconds
[2023-08-29T22:04:33.438+0000] {processor.py:157} INFO - Started process (PID=24230) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:04:33.439+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:04:33.441+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:04:33.441+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:04:33.479+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:04:33.471+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:04:33.480+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:04:33.541+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.106 seconds
[2023-08-29T22:05:04.021+0000] {processor.py:157} INFO - Started process (PID=24325) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:05:04.022+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:05:04.023+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:05:04.023+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:05:04.058+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:05:04.051+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:05:04.058+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:05:04.095+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.077 seconds
[2023-08-29T22:05:34.682+0000] {processor.py:157} INFO - Started process (PID=24420) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:05:34.684+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:05:34.686+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:05:34.686+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:05:34.734+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:05:34.725+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:05:34.735+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:05:34.775+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.098 seconds
[2023-08-29T22:06:05.081+0000] {processor.py:157} INFO - Started process (PID=24515) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:06:05.083+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:06:05.084+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:06:05.084+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:06:05.117+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:06:05.110+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:06:05.118+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:06:05.168+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.089 seconds
[2023-08-29T22:06:35.606+0000] {processor.py:157} INFO - Started process (PID=24610) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:06:35.607+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:06:35.608+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:06:35.608+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:06:35.638+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:06:35.632+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:06:35.639+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:06:35.674+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.073 seconds
[2023-08-29T22:07:06.074+0000] {processor.py:157} INFO - Started process (PID=24705) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:07:06.075+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:07:06.076+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:07:06.076+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:07:06.109+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:07:06.103+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:07:06.109+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:07:06.145+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.074 seconds
[2023-08-29T22:07:36.710+0000] {processor.py:157} INFO - Started process (PID=24800) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:07:36.711+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:07:36.712+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:07:36.712+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:07:36.743+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:07:36.737+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:07:36.743+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:07:36.779+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.074 seconds
[2023-08-29T22:08:07.399+0000] {processor.py:157} INFO - Started process (PID=24895) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:08:07.400+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:08:07.401+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:08:07.401+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:08:07.435+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:08:07.428+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:08:07.435+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:08:07.485+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.091 seconds
[2023-08-29T22:08:38.441+0000] {processor.py:157} INFO - Started process (PID=24990) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:08:38.443+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:08:38.444+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:08:38.444+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:08:38.476+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:08:38.470+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:08:38.477+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:08:38.511+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.072 seconds
[2023-08-29T22:09:08.994+0000] {processor.py:157} INFO - Started process (PID=25085) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:09:08.995+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:09:08.996+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:09:08.996+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:09:09.026+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:09:09.020+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:09:09.027+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:09:09.063+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.072 seconds
[2023-08-29T22:09:39.454+0000] {processor.py:157} INFO - Started process (PID=25180) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:09:39.455+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:09:39.457+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:09:39.457+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:09:39.494+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:09:39.487+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:09:39.495+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:09:39.530+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.081 seconds
[2023-08-29T22:10:09.910+0000] {processor.py:157} INFO - Started process (PID=25275) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:10:09.911+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:10:09.912+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:10:09.912+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:10:09.946+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:10:09.939+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:10:09.947+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:10:10.000+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.093 seconds
[2023-08-29T22:10:40.294+0000] {processor.py:157} INFO - Started process (PID=25370) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:10:40.295+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:10:40.296+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:10:40.296+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:10:40.327+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:10:40.321+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:10:40.328+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:10:40.363+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.073 seconds
[2023-08-29T22:11:11.089+0000] {processor.py:157} INFO - Started process (PID=25465) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:11:11.090+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:11:11.091+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:11:11.091+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:11:11.123+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:11:11.117+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:11:11.123+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:11:11.175+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.089 seconds
[2023-08-29T22:11:41.948+0000] {processor.py:157} INFO - Started process (PID=25560) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:11:41.950+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:11:41.951+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:11:41.950+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:11:41.980+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:11:41.975+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:11:41.981+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:11:42.018+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.072 seconds
[2023-08-29T22:12:12.558+0000] {processor.py:157} INFO - Started process (PID=25655) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:12:12.560+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:12:12.563+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:12:12.562+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:12:12.612+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:12:12.604+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:12:12.613+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:12:12.650+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.100 seconds
[2023-08-29T22:12:43.153+0000] {processor.py:157} INFO - Started process (PID=25755) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:12:43.154+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:12:43.155+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:12:43.155+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:12:43.184+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:12:43.179+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:12:43.185+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:12:43.219+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.069 seconds
[2023-08-29T22:13:14.057+0000] {processor.py:157} INFO - Started process (PID=25850) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:13:14.058+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:13:14.059+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:13:14.059+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:13:14.089+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:13:14.083+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:13:14.090+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:13:14.127+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.074 seconds
[2023-08-29T22:13:44.356+0000] {processor.py:157} INFO - Started process (PID=25945) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:13:44.357+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:13:44.359+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:13:44.358+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:13:44.397+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:13:44.389+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:13:44.397+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:13:44.436+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.083 seconds
[2023-08-29T22:14:14.975+0000] {processor.py:157} INFO - Started process (PID=26040) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:14:14.976+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:14:14.978+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:14:14.978+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:14:15.013+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:14:15.006+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:14:15.013+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:14:15.103+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.132 seconds
[2023-08-29T22:14:45.448+0000] {processor.py:157} INFO - Started process (PID=26135) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:14:45.449+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:14:45.450+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:14:45.450+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:14:45.483+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:14:45.477+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:14:45.484+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:14:45.527+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.082 seconds
[2023-08-29T22:15:16.307+0000] {processor.py:157} INFO - Started process (PID=26230) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:15:16.308+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:15:16.310+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:15:16.309+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:15:16.345+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:15:16.337+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:15:16.346+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:15:16.384+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.081 seconds
[2023-08-29T22:15:47.180+0000] {processor.py:157} INFO - Started process (PID=26325) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:15:47.181+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:15:47.183+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:15:47.182+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:15:47.218+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:15:47.211+0000] {dagbag.py:350} ERROR - Failed to import: /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 346, in parse
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/airflow/dags/dags_python_with_trigger_rule_eg2.py", line 44, in <module>
    random_branch() >> [task_a, task_b(), task_c()] >> task_d()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 345, in __call__
    **self.kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/python.py", line 54, in __init__
    **kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/decorators/base.py", line 212, in __init__
    super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 166, in __init__
    super().__init__(**kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 429, in apply_defaults
    result = func(self, **kwargs, default_args=default_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 864, in __init__
    f"The trigger_rule must be one of {TriggerRule.all_triggers()},"
airflow.exceptions.AirflowException: The trigger_rule must be one of {<TriggerRule.ALL_DONE: 'all_done'>, <TriggerRule.ALL_FAILED: 'all_failed'>, <TriggerRule.ALL_SKIPPED: 'all_skipped'>, <TriggerRule.ONE_SUCCESS: 'one_success'>, <TriggerRule.NONE_FAILED_MIN_ONE_SUCCESS: 'none_failed_min_one_success'>, <TriggerRule.ONE_FAILED: 'one_failed'>, <TriggerRule.NONE_FAILED: 'none_failed'>, <TriggerRule.NONE_FAILED_OR_SKIPPED: 'none_failed_or_skipped'>, <TriggerRule.DUMMY: 'dummy'>, <TriggerRule.NONE_SKIPPED: 'none_skipped'>, <TriggerRule.ALL_SUCCESS: 'all_success'>, <TriggerRule.ONE_DONE: 'one_done'>, <TriggerRule.ALWAYS: 'always'>},'dags_python_with_trigger_rule_eg2.task_d'; received 'non_skipped'.
[2023-08-29T22:15:47.218+0000] {processor.py:838} WARNING - No viable dags retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:15:47.338+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.161 seconds
[2023-08-29T22:16:11.815+0000] {processor.py:157} INFO - Started process (PID=26421) to work on /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:16:11.816+0000] {processor.py:826} INFO - Processing file /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py for tasks to queue
[2023-08-29T22:16:11.817+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:16:11.817+0000] {dagbag.py:541} INFO - Filling up the DagBag from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:16:11.866+0000] {processor.py:836} INFO - DAG(s) dict_keys(['dags_python_with_trigger_rule_eg2']) retrieved from /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py
[2023-08-29T22:16:12.027+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:16:12.027+0000] {manager.py:504} INFO - Created Permission View: can edit on DAG:dags_python_with_trigger_rule_eg2
[2023-08-29T22:16:12.059+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:16:12.058+0000] {manager.py:504} INFO - Created Permission View: can read on DAG:dags_python_with_trigger_rule_eg2
[2023-08-29T22:16:12.084+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:16:12.084+0000] {manager.py:504} INFO - Created Permission View: can delete on DAG:dags_python_with_trigger_rule_eg2
[2023-08-29T22:16:12.102+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:16:12.102+0000] {dag.py:2742} INFO - Sync 1 DAGs
[2023-08-29T22:16:12.117+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:16:12.117+0000] {dag.py:2763} INFO - Creating ORM DAG for dags_python_with_trigger_rule_eg2
[2023-08-29T22:16:12.130+0000] {logging_mixin.py:150} INFO - [2023-08-29T22:16:12.130+0000] {dag.py:3508} INFO - Setting next_dagrun for dags_python_with_trigger_rule_eg2 to None, run_after=None
[2023-08-29T22:16:12.238+0000] {processor.py:179} INFO - Processing /opt/airflow/dags/dags_python_with_trigger_rule_eg2.py took 0.428 seconds
